modules/api/test/LorisApiDicomsTest.php

```diff
@@ -0,0 +1,375 @@
+ * of the api module and look at the response content, status code and headers where
+ * it applies. All endpoints are accessible at <host>/api/<version>/
+ * (e.g. the endpoint of the version 0.0.3 ofd the API "/projects" URI for the host
+ * "example.loris.ca" would be https://example.loris.ca/api/v0.0.3/projects)
+ *
+ * @category   API
+ * @package    Tests
+ * @subpackage Integration
+ * @author     Simon Pelletier <simon.pelletier@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class LorisApiDicomsTests extends LorisApiTests
+{
+    /**
+     * Tests the HTTP GET request for the
+     * endpoint /candidates/{candid}/{visit}/dicoms
+     *
+     * @return void
+     */
+    public function testGetCandidatesCandidVisitDicoms(): void
+    {
+        $this->guzzleLogin();
+        $response     = $this->client->request(
+            'GET',
+            "$this->base_uri/candidates",
+            [
+                'headers' => $this->headers
+            ]
+        );
+        $candidsArray = json_decode(
+            (string) utf8_encode(
+                $response->getBody()->getContents()
+            ),
+            true
+        );
+        $candids      = array_keys($candidsArray['Candidates']);
+        foreach ($candids as $candid) {
+            $id          = $candidsArray['Candidates'][$candid]['CandID'];
+            $response    = $this->client->request(
+                'GET',
+                "$this->base_uri/candidates/$id",
+                [
+                    'headers' => $this->headers
+                ]
+            );
+            $visitsArray = json_decode(
+                (string) utf8_encode(
+                    $response->getBody()->getContents()
+                ),
+                true
+            );
+            $visits      = array_keys($visitsArray['Visits']);
+            foreach ($visits as $visit) {
+                $v        = $visitsArray['Visits'][$visit];
+                $response = $this->client->request(
+                    'GET',
+                    "$this->base_uri/candidates/$id/$v/dicoms/",
+                    [
+                        'headers' => $this->headers
+                    ]
+                );
+                $this->assertEquals(200, $response->getStatusCode());
+                $headers = $response->getHeaders();
+                $this->assertNotEmpty($headers);
+                foreach ($headers as $header) {
+                    $this->assertNotEmpty($header);
+                    //$this->assertIsString($header[0]);
+                }
+                // Verify the endpoint has a body
+                $body = $response->getBody();
+                $this->assertNotEmpty($body);
+            }
+        }
+    }
+
+    /**
+     * Tests the HTTP POST request for the
+     * endpoint /candidates/{candid}/{visit}/dicoms
+     * // TODO HANDLING OF POST HANDLING NOT IMPLEMENTED
```

Comment:
It can test that the response is 405 Method not allowed.

Reply:
it is not implemented yet, so I tested for 404 not found